From 25c75dcc1d911c3b97231e22fffa94da7bccb2bd Mon Sep 17 00:00:00 2001
From: Konstantin Pavlov <thresh@nginx.com>
Date: Wed, 22 Nov 2023 13:46:29 -0800
Subject: [PATCH] Fixed build with GCC 4.8.5 on RHEL 7.

---
 lib/cpp/src/thrift/transport/TBufferTransports.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/cpp/src/thrift/transport/TBufferTransports.h b/lib/cpp/src/thrift/transport/TBufferTransports.h
index 8518800a4..e74143243 100644
--- a/lib/cpp/src/thrift/transport/TBufferTransports.h
+++ b/lib/cpp/src/thrift/transport/TBufferTransports.h
@@ -23,6 +23,7 @@
 #include <cstdlib>
 #include <cstring>
 #include <limits>
+#include <cstddef>
 
 #include <thrift/transport/TTransport.h>
 #include <thrift/transport/TVirtualTransport.h>
-- 
2.46.1

