From 28f0acce2ebcd20f236e2d804a2ebef77b0f5640 Mon Sep 17 00:00:00 2001
From: Andrei Belov <defanator@gmail.com>
Date: Sat, 22 Dec 2018 02:39:31 +0300
Subject: [PATCH] Fixed building with gcc 8

---
 ngx_rtmp_eval.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ngx_rtmp_eval.c b/ngx_rtmp_eval.c
index 1e5195a9..c658f14e 100644
--- a/ngx_rtmp_eval.c
+++ b/ngx_rtmp_eval.c
@@ -166,6 +166,7 @@ ngx_rtmp_eval(void *ctx, ngx_str_t *in, ngx_rtmp_eval_t **e, ngx_str_t *out,
                         state = ESCAPE;
                         continue;
                 }
+                /* fall through */
 
             case ESCAPE:
                 ngx_rtmp_eval_append(&b, &c, 1, log);
