diff --git a/src/nginx_module/Configuration.c b/src/nginx_module/Configuration.c
index 33a16cbf8..737f21598 100644
--- a/src/nginx_module/Configuration.c
+++ b/src/nginx_module/Configuration.c
@@ -224,7 +224,9 @@ passenger_create_loc_conf(ngx_conf_t *cf)
 
     conf->upstream_config.send_lowat = NGX_CONF_UNSET_SIZE;
     conf->upstream_config.buffer_size = NGX_CONF_UNSET_SIZE;
-    #if NGINX_VERSION_NUM >= 1007007
+    #if NGINX_VERSION_NUM >= 1027000
+        conf->upstream_config.limit_rate = NGX_CONF_UNSET_PTR;
+    #elif NGINX_VERSION_NUM >= 1007007
         conf->upstream_config.limit_rate = NGX_CONF_UNSET_SIZE;
     #endif
 
@@ -465,7 +467,10 @@ passenger_merge_loc_conf(ngx_conf_t *cf, void *parent, void *child)
                               prev->upstream_config.buffer_size,
                               16 * 1024);
 
-    #if NGINX_VERSION_NUM >= 1007007
+    #if NGINX_VERSION_NUM >= 1027000
+        ngx_conf_merge_ptr_value(conf->upstream_config.limit_rate,
+                                  prev->upstream_config.limit_rate, NULL);
+    #elif NGINX_VERSION_NUM >= 1007007
         ngx_conf_merge_size_value(conf->upstream_config.limit_rate,
                                   prev->upstream_config.limit_rate, 0);
     #endif
