diff --git a/others/Makefile.am b/others/Makefile.am
index 7501d558..f7c9f177 100644
--- a/others/Makefile.am
+++ b/others/Makefile.am
@@ -5,6 +5,8 @@ libinjection_la_SOURCES = \
 	libinjection/src/libinjection_sqli.c \
 	libinjection/src/libinjection_xss.c
 
+libinjection_la_CFLAGS = -DLIBINJECTION_VERSION=\"@@LIBINJECTION_VERSION@@\"
+
 noinst_HEADERS = \
 	libinjection/src/libinjection.h \
 	libinjection/src/libinjection_html5.h \
